codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketChunkedInput.java

```diff
@@ -0,0 +1,113 @@
+ * At the end of the input data, {@link ContinuationWebSocketFrame} with finalFragment will be written.
+ * <p>
+ */
+public class WebSocketChunkedInput implements ChunkedInput<WebSocketFrame> {
+
+    private final ChunkedInput<ByteBuf> input;
+    private final ContinuationWebSocketFrame lastWebSocketContent;
+    private boolean sentLastChunk;
+    private final int rsv;
+
+    /**
+     * Creates a new instance using the specified input.
+     * @param input {@link ChunkedInput} containing data to write
+     */
+    public WebSocketChunkedInput(ChunkedInput<ByteBuf> input) {
+        this.input = input;
+        this.lastWebSocketContent = new ContinuationWebSocketFrame();
+        this.sentLastChunk = false;
+        this.rsv = 0;
+
+    }
+
+    /**
+     * Creates a new instance using the specified input.
+     * @param input {@link ChunkedInput} containing data to write
+     * @param lastWebSocketContent {@link ContinuationWebSocketFrame} that will be written as the terminating chunk.
+     *            Use this for training headers.
+     * @param rsv RSV1, RSV2, RSV3 used for extensions
+     */
+    public WebSocketChunkedInput(ChunkedInput<ByteBuf> input,
+                                 ContinuationWebSocketFrame lastWebSocketContent,
+                                 int rsv) {
+        this.input = input;
+        this.lastWebSocketContent = lastWebSocketContent;
+        this.sentLastChunk = false;
+        this.rsv = rsv;
+    }
+
+    /**
+     * @return {@code true} if and only if there is no data left in the stream
+     * and the stream has reached at its end.
+     */
+    @Override
+    public boolean isEndOfInput() throws Exception {
+        if (input.isEndOfInput()) {
+            // Only end of input after last WebSocket frame chunk has been sent
+            return sentLastChunk;
+        } else {
+            return false;
+        }
+    }
+
+
+    /**
+     * Releases the resources associated with the input.
+     */
+    @Override
+    public void close() throws Exception {
+        input.close();
+    }
+
+    /**
+     * Fetches a chunked data from the stream. Once this method returns the last chunk
+     * and thus the stream has reached at its end, any subsequent {@link #isEndOfInput()}
+     * call must return {@code true}.
+     *
+     * @param ctx {@link ChannelHandlerContext} context of channelHandler
+     * @return {@link WebSocketFrame} contain chunk of data
+     */
+    @Override
+    public WebSocketFrame readChunk(ChannelHandlerContext ctx) throws Exception {
+        if (input.isEndOfInput()) {
+            if (sentLastChunk) {
+                return null;
+            } else {
+                // Send last chunk for this input
+                sentLastChunk = true;
+                return lastWebSocketContent;
+            }
+        } else {
+            return new ContinuationWebSocketFrame(false, rsv, input.readChunk(ctx));
```

Comment:
@normanmaurer 
i mistake to check return null value because i didn't seen the ChunkedWriteHandler class convert null value to empty buffer  (Unpooled.EMPTY_BUFFER ) then write to channel.